### ISSUE TITLE
Recursive class doesn't even expand

### ISSUE THREAD

**Issue by igarashitm** [atlasmap/atlasmap#255]
The fix for #81 finally revealed the "right" issue, recursive class can't expand in UI. UI doesn't support expanding cached class. Now `/quotedStats` doesn't expand as it's recursive twitter4j.Status class.
![screenshot from 2018-01-26 16-43-33](https://user-images.githubusercontent.com/265462/35462179-e3fbaecc-02b8-11e8-8aa3-4ddd9f533066.png)

We should
* Gray out the recursive(cached) field until expand
* Un-gray out and show fields once it's expanded

**Comment by rohanmars** [comment]
It might be a different issue, but I think it's related, but referencing the same type form multiple times in an object wont populate (or automatically display) the second instance sub fields of that type. The UI should probably pre populate the cache based on the cached values, so that search works across repeating fields. It would have a potential impact on performance, but the priming could be capped at some magic field count.

Greying out recursive fields is a good idea!

**Comment by igarashitm** [comment]
Yep, sounds like a same root cause. Updated the title and description.

**Comment by pleacu** [comment]
![i255](https://user-images.githubusercontent.com/2981421/36104937-5a80a70c-0fe1-11e8-9a93-eb2918d316af.png)
Kept the expansion arrow so users can expand and have content filled in when un-caches.

**Comment by jpav** [comment]
Seems like we should also have a tooltip that explains why the field is grayed out, i.e., that it's recursive.
